include/nbl/system/IAsyncQueueDispatcher.h

```diff
@@ -15,63 +15,79 @@ namespace impl
     public:
         IAsyncQueueDispatcherBase() = default;
         ~IAsyncQueueDispatcherBase() = default;
+        // dont want to play around with relaxed memory ordering yet
         struct request_base_t
         {
+                enum E_STATE : uint32_t
+                {
+                    ES_INITIAL=0,
+                    ES_RECORDING=1,
+                    ES_PENDING=2,
+                    ES_EXECUTING=3,
+                    ES_READY=4
+                };
+                request_base_t() : state(ES_INITIAL)
+                {
+                    assert(std::atomic_uint32_t::is_lock_free());
```

Comment:
are the platforms where this could not be true? out of curiosity

Reply:
dunno, RISC-V, some embedded CPUs?